torchgeo/datasets/oscd.py

```diff
@@ -0,0 +1,308 @@
+
+    Dataset classes:
+
+    1. no change
+    2. change
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1810.08468
+    """
+
+    url = "https://drive.google.com/file/d/1jidN0DKEIybOrP0j7Bos8bGDDq3Varj3"
+    md5 = "7383412da7ece1dca1c12dc92ac77f09"
+
+    zipfile_glob = "*OSCD.zip"
+    zipfile_glob2 = "*Onera*.zip"
+    filename_glob = "*Onera*"
+    filename = "OSCD.zip"
+    splits = ["train", "test"]
+
+    colormap = ["blue"]
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new OSCD dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        assert split in self.splits
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        self.files = self._load_files()
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        # TODO: implement choosing bands (right now assuming bands="all")
+        image1 = self._load_image(files["images1"])
+        image2 = self._load_image(files["images2"])
+        mask = self._load_target(str(files["mask"]))
+
+        image = torch.stack(tensors=[image1, image2], dim=0)
+        sample = {"image": image, "mask": mask}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self) -> List[Dict[str, Union[str, Sequence[str]]]]:
+        regions = []
+        temp_split = "Test" if self.split == "test" else "Train"
+        labels_root = os.path.join(
+            self.root, f"Onera Satellite Change Detection dataset - {temp_split} Labels"
```

Comment:
You can use `self.split.capitalize` instead of creating a new variable `temp_split`